adapter/src/test/kotlin/de/fraunhofer/iem/spha/adapter/ToolResultParserTest.kt

```diff
@@ -0,0 +1,207 @@
+    }
+
+    @Test
+    fun testParseJsonFilesFromDirectoryWithTrivyFile() {
+        // Copy only the Trivy file to a temporary directory
+        val tempTrivyDir = createTempDirectory("trivy")
+        val trivyFile = File("$testResourcesDir/trivy-result-v2.json")
+        val tempTrivyFile = tempTrivyDir.resolve("trivy-result-v2.json").toFile()
+        trivyFile.copyTo(tempTrivyFile)
+
+        // We're just testing that the file is processed without errors
+        // The actual results might be empty depending on the adapter implementation
+        ToolResultParser.parseJsonFilesFromDirectory(tempTrivyDir.toString())
+
+        // Clean up
+        tempTrivyDir.toFile().deleteRecursively()
+    }
+
+    @Test
+    fun testParseJsonFilesFromDirectoryWithTrufflehogFile() {
+        // Copy only the Trufflehog file to a temporary directory
+        val tempTrufflehogDir = createTempDirectory("trufflehog")
+        val trufflehogFile = File("$testResourcesDir/trufflehog.json")
+        val tempTrufflehogFile = tempTrufflehogDir.resolve("trufflehog.json").toFile()
+        trufflehogFile.copyTo(tempTrufflehogFile)
+
+        // We're just testing that the file is processed without errors
+        // The actual results might be empty depending on the adapter implementation
+        ToolResultParser.parseJsonFilesFromDirectory(tempTrufflehogDir.toString())
+
+        // Clean up
+        tempTrufflehogDir.toFile().deleteRecursively()
+    }
+
+    @Test
+    fun testParseJsonFilesFromDirectoryWithTlcFile() {
+        // Copy only the TLC file to a temporary directory
+        val tempTlcDir = createTempDirectory("tlc")
+        val tlcFile = File("$testResourcesDir/tlc-result-npm.json")
+        val tempTlcFile = tempTlcDir.resolve("tlc-result-npm.json").toFile()
+        tlcFile.copyTo(tempTlcFile)
+
+        val results = ToolResultParser.parseJsonFilesFromDirectory(tempTlcDir.toString())
+
+        // Clean up
+        tempTlcDir.toFile().deleteRecursively()
+
+        assertTrue(results.isNotEmpty())
+        assertTrue(results.all { it is AdapterResult.Success })
+    }
+
+    @Test
+    fun testParseJsonFilesFromDirectoryWithInvalidJsonFile() {
+        val results = ToolResultParser.parseJsonFilesFromDirectory(invalidJsonDir.toString())
+        assertTrue(results.isEmpty())
+    }
+
+    @Test
+    fun testParseJsonFilesFromDirectoryWithEmptyJsonFile() {
+        // Create an empty JSON file
+        val emptyJsonDir = createTempDirectory("empty-json")
+        val emptyJsonFile = emptyJsonDir.resolve("empty.json").toFile()
+        emptyJsonFile.writeText("")
+
+        val results = ToolResultParser.parseJsonFilesFromDirectory(emptyJsonDir.toString())
+
+        // Clean up
+        emptyJsonDir.toFile().deleteRecursively()
+
+        assertTrue(results.isEmpty())
+    }
+
+    @Test
+    fun testGetAdapterResultsFromJsonFiles() {
+        val jsonFiles =
+            listOf(
+                File("$testResourcesDir/osv-scanner.json"),
```

Comment:
vary the writing of `.json` cause you have enabled ignore case, which i consinder to be "contracted behavior"